test/adaptors/in-memory-articles.test.ts

```diff
@@ -1,116 +1,126 @@
-import { Iri, JsonLdObj } from 'jsonld/jsonld-spec';
+import { blankNode, literal, quad } from '@rdfjs/data-model';
+import { equals } from 'rdf-dataset-ext';
+import { BlankNode, DatasetCore } from 'rdf-js';
 import InMemoryArticles from '../../src/adaptors/in-memory-articles';
 import ArticleNotFound from '../../src/errors/article-not-found';
-import ArticleHasNoId from '../../src/errors/article-has-no-id';
 import NotAnArticle from '../../src/errors/not-an-article';
+import { rdf, schema } from '../../src/namespaces';
 import createArticle from '../create-article';
 
 describe('in-memory articles', (): void => {
   it('can add an article', async (): Promise<void> => {
     const articles = new InMemoryArticles();
+    const id = blankNode();
 
-    expect(await articles.contains('_:1')).toBe(false);
+    expect(await articles.contains(id)).toBe(false);
 
-    await articles.add(createArticle('_:1'));
+    await articles.set(id, createArticle(id));
 
-    expect(await articles.contains('_:1')).toBe(true);
+    expect(await articles.contains(id)).toBe(true);
   });
 
   it('can add an article with multiple types', async (): Promise<void> => {
     const articles = new InMemoryArticles();
+    const id = blankNode();
 
-    await articles.add({
-      ...createArticle('_:1'),
-      '@type': ['http://schema.org/Article', 'http://schema.org/NewsArticle'],
-    });
+    await articles.set(id, createArticle(id).add(quad(id, rdf.type, schema.NewsArticle)));
 
-    expect(await articles.contains('_:1')).toBe(true);
+    expect(await articles.contains(id)).toBe(true);
   });
 
   it('can update an article', async (): Promise<void> => {
     const articles = new InMemoryArticles();
+    const id = blankNode();
 
-    await articles.add(createArticle('_:1', 'Original'));
-    await articles.add(createArticle('_:1', 'Updated'));
+    await articles.set(id, createArticle(id, literal('Original')));
+    await articles.set(id, createArticle(id, literal('Updated')));
 
-    expect((await articles.get('_:1'))['http://schema.org/name']).toBe('Updated');
+    expect((await articles.get(id)).has(quad(id, schema.name, literal('Updated')))).toBe(true);
   });
 
   it('throws an error if it is not an article', async (): Promise<void> => {
     const articles = new InMemoryArticles();
+    const id = blankNode();
 
-    await expect(articles.add({
-      ...createArticle('_:1'),
-      '@type': 'http://schema.org/NewsArticle',
-    })).rejects.toThrow(new NotAnArticle(['http://schema.org/NewsArticle']));
+    const article = createArticle(id)
+      .delete(quad(id, rdf.type, schema.Article))
+      .add(quad(id, rdf.type, schema.NewsArticle));
```

Comment:
pass `schema.NewsArticle` as an optional argument rather than deleting the other type?